### ISSUE TITLE
add backup account / recovery code step to signup

### ISSUE THREAD

**Issue by rdevries-thoughtworks** [pixelated/pixelated-user-agent#802]
_spawned from #791, depends on #801, estimation: 16 points_ 

Create a screen for entering backup email account. On submit, **pixelated** generates a recovery code, e.g. in the same way the Soledad secret is created, and sets it as the recovery_code on both Soledad and Bonafide. To do this, setting a recovery_code should be added to both Soledad and Bonafide (LEAP pull requests).

For **Soledad**, we should add (something like) `Soledad.create_recovery_code()` to LEAP's Soledad client. This method:

- generates a random recovery code R in the Soledad client
- encrypts the Soledad secret with R (first decrypts using the password)
- adds that encrypted secret to the `recovery_document`

For **Bonafide**, we should augment the `update user` API to allow setting the `recovery_code_verifier` (+ salt). This affects the Bonafide server (RoR web app) and client (Python).

The code R is split into a user and an admin part by **pixelated**. These parts have to be combined to get the recovery code. The user part is sent to the backup email address, the admin part to the admin's Pixelated account, which is set through a command line parameter (for now). Probably [ssss](https://en.wikipedia.org/wiki/Shamir%27s_Secret_Sharing) is the way to go here. Currently, 1 admin part and the user part are needed to get the recovery code, but in the future this might change (see [this comment](https://github.com/pixelated/pixelated-user-agent/issues/791#issuecomment-253246965)). The code R can be created in the same way as the Soledad secret.

**Acceptance criteria**
- reloading Pixelated when there's no backup email yet shows the backup account screen
- Pixelated doesn't keep track of the email address
- on startup, fail if the (configured) admin email address doesn't exists on Pixelated

**Tasks**
- [ ] add `Soledad.create_recovery_code()` to LEAP's Soledad client _[2 points]_
- [ ] allow setting the recovery code on the `update user` API of the Bonafide server _[5 points]_
- [ ] add `Session.create_recovery_code()` to LEAP's Bonafide client _[1 point]_
- [ ] add Pixelated front-end & send parts of the recovery code to user and admin _[8 points]_

**Comment by olabini** [comment]
I'm a bit confused by this issue. It doesn't seem to take into account the splitting of thecode.
( because it says to mail R to the user )

**Comment by rdevries-thoughtworks** [comment]
I've created a separate story for splitting. The reason being that smaller stories can be prioritized and estimated separately, and maybe even parallelized.

**Comment by olabini** [comment]
Well, splitting is good, but each smaller story needs to have business value by itself - otherwise it's not a story. Mailing R in whole is dangerous and not a good idea at all, so that's not a valid way of splitting the story.

**Comment by rdevries-thoughtworks** [comment]
Agreed, I closed #803 and included it in this issue. I've not merged #804 though.

**Comment by rdevries-thoughtworks** [comment]
Closing this issue because it's split into #872, #873, #874, #875 and #876.
